docs/DeveloperGuide.md

```diff
@@ -949,6 +1267,162 @@ Although all fields can be edited, the `Name`, `Amount`, and `Type` fields canno
+the lifeline reaches the end of diagram.
+
+**Details**
+
+1. The user specifies the transaction to be retrieved by stating the integer index of the transaction to be retrieved.
+2. The input will be parsed by `GetCommandParser` and if the provided input is invalid, `ParseException` will be thrown,
+   and the user is prompted to enter the command again with the correct input.
+3. If the input is valid, an `Index` object is created with the given input integer, and passed into `GetCommand` to be executed
+   by `LogicManager`
+4. `LogicManger` will invoke the `execute` method of `GetCommand` which will retrieve the `Transaction` from UniCa$h.
+
+It is important to take note that when the user input is parsed, it is based on the currently displayed `TransactionList` inside
+`TransactionListPanel`. For more information, you can refer back to how [DeleteTransactionCommand]() handles indices.
+
+#### Reset UniCa$h
+
+**Overview**
+
+The `reset_unicash` command overwrites all existing `Transactions` in `TransactionList`
+with the default UniCa$h `Transactions` 
+
+The activity diagram of resetting UniCa$h is as shown below
+
+<img src="images/unicash/diagrams/ResetUniCashActivityDiagram.png" width="450" />
+
+The following sequence diagram shows the interaction between different components of UniCash.
+
+<img src="images/unicash/diagrams/ResetUniCashSequenceDiagram.png" width="800" />
+
+
+**Details**
+
+1. The user inputs the command to clear all transactions
+2. The input will be parsed by `ResetCommandParser` and if the provided input is
+   invalid (i.e. trailing arguments) `ParseException` will be thrown, and the
+   user is prompted to enter the command again with the correct input.
+3. If the input is valid, a `ResetCommand` object is created to be executed by `LogicManager`
+4. `LogicManager` will invoke the `execute` method of `ResetCommand`
+   which will replace the existing `Model` property with a sample `UniCash` object which
+   would contain a `TransactionList` populated with sample `Transaction` data.
+
+<div class="callout callout-info" markdown="span">
+It must be noted here that resetting UniCa$h to its original state refers to overwriting
+the internal `Transactions List` with the original transactions only, and does not refer to
+clearing any contained `budgets`
+</div>
+
+#### Find Transactions
+
+**Overview**
+
+The `find` command searches UniCa$h for `Transactions` that match the input parameters.
+
+The activity diagram of finding transactions in UniCa$h is as shown below
+
+<img src="images/unicash/diagrams/FindTransactionsActivityDiagram.png" width="400" />
+
+The following sequence diagram shows the interaction between different components of UniCa$h
+during the sequence of processing and execution of the `find` command
+
+<img src="images/unicash/diagrams/FindTransactionsSequenceDiagram.png" width="800" />
```

Comment:
I can't comment on the image directly, but is `FindCommandParser` supposed to terminate after its activation bar?

Reply:
Nice catch, thanks for noticing, I've gone ahead and changed that.